test/EncDecTest.cpp

```diff
@@ -1269,11 +1270,18 @@ test_optional()
 	fail_unless(ok);
 	fail_unless(!opt_num.has_value());
 
+	ok = mpp::decode(run, opt_num);
+	fail_unless(ok);
+	fail_unless(opt_num.has_value());
+	fail_unless(opt_num.value() == 42);
+
 	buf.flush();
 
 	TEST_CASE("containers with numbers");
 	int null_idx = 4;
-	mpp::encode(buf, mpp::as_arr(std::forward_as_tuple(0, 1, 2, 3, nullptr, 5)));
+	mpp::encode(buf, std::make_optional(mpp::as_arr(
+		std::forward_as_tuple(0, std::make_optional(1), 2, 3, std::optional<int>(), 5)
+	)));
```

Comment:
Nit: no sure it's worth it, but I think adding separate test cases would be cleaner than wrapping around existing ones.

Same comment for `objects` test case below.